### ISSUE TITLE
Configure NGINX to load multiple projects based on domain automatically

### ISSUE THREAD

**Issue by javiertoledos** [laradock/laradock#409]
It would be nice to enable loading multiple projects based on the domain used without having to add extra config files. Instead of manually creating each projectname.conf file inside nginx/sites/ you can open the project site by typing in your browser "project1.laravel.dev" and mapping it to ../$project1 folder. In this way, you don't even need to add an extra volume or nginx conf for multiple projects. 

Here is an example of how this would work:

```
server {
    server_name ~^(?<project>.+)\.laradock\.dev$;
    root /var/www/$project;
}
```
So, when visiting project1.laradock.dev, the root folder becomes ../project1, when visiting project2.laradock.dev, it becomes ../project2, and so on. 

An additional  enhancement to this feature would be to load different platforms configurations based on the used domain besides Laravel, so we would have additional server blocks with this kind of configuration:

```
server { 
    server_name ~^(?<project>.+)\.wordpress\.dev$;
    root /var/www/$project;

    #NGINX Wordpress specific config goes here...
}
```

```
server { 
    server_name ~^(?<project>.+)\.drupal8\.dev$;
    root /var/www/$project;

    #NGINX D8 specific config goes here...
}
```

Basically, depending on the domain used, you get an NGINX configuration tailored for that platform. This feature can be disabled  by default and enable it in the docker.compose.yml with an argument:

```
    nginx:
        build:
            context: ./nginx
            args:
                # Arg name suggestions are welcome
                - LOAD_PROJECTS_WITH_DEV_DOMAINS=true
```

Also, a guide (or references) can be added explaining how to configure dnsmasq (Unix based systems) or Acrylic ( Windows) for resolving any *.dev domains in order to get a nice multiplatform PHP dev environment.

If this is something you're interested in, I can work on it and do a Pull Request.

**Comment by Mahmoudz** [comment]
@javiertoledos that looks great solution, just my concern is does this work with subdomains?!
say my website has these 2 domains `api.website.dev` and `admin.website.dev` will they work with the setup above? Note the current solution is not working with subdomains for some reason!

If so I'd love to get PR from you.

Best,

**Comment by manshoor** [comment]
Hey @Mahmoudz,

I have been use the subdomain like this on Laradock 
`server_name la.local *.la.local;` and it's working perfectly fine.
One thing though Laradock is extremely slow (5x slow than homestead) on my side and because of it I'm not able to use it properly

**Comment by Mahmoudz** [comment]
@manshoor thanks for sharing I'll check that again, I have to say I didn't give enough time to try..

Well yes the speed is a problem, please check this issue https://github.com/laradock/laradock/issues/353 and hopfully we solve this soon if it's Laradock problem.
